### ISSUE TITLE
[mtt] segfault in mpich_tests_mpi_comm

### ISSUE THREAD

**Issue by amaslenn** [openucx/ucx#2664]
Configuration
```
OMPI: 3.1.1a1
MOFED: MLNX_OFED_LINUX-4.3-1.0.1.0
Nodes: hercules x16 (ppn=32(x16), nodelist=clx-hercules-[030-031,041-042,060-061,067-068,102-109])
Job: ucx
```

MTT log: http://e2e-gw.mellanox.com:4080//mnt/lustre/users/mtt/scratch/ucx_ompi/20180603_083232_32203_107955_clx-hercules-030/html/test_stdout_6OKiqJ.txt

Cmd:
`mpirun -np 512 --display-map -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to core -x UCX_NET_DEVICES=mlx5_0:1,mlx5_2:1 -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=rc_x,sm -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 --map-by node /mnt/lustre/users/mtt/scratch/ucx_ompi/20180603_083232_32203_107955_clx-hercules-030/installs/N__o/tests/mpich_tests/mpich-mellanox.git/test/mpi/comm/comm_idup`

Output:
```sh
=============================================================
Sun Jun  3 10:44:33 2018[1,0]<stdout>: No errors
Sun Jun  3 10:44:38 2018[1,364]<stderr>:[clx-hercules-106:4124 :0:4301] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
Sun Jun  3 10:44:38 2018[1,364]<stderr>:==== backtrace ====
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 0 0x000000000001fce0 uct_ep_get_address()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/uct/base/uct_iface.c:483
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 1 0x00000000000319aa ucp_wireup_connect_local()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/ucp/wireup/wireup.c:193
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 2 0x0000000000032ac8 ucp_wireup_msg_handler()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/ucp/wireup/wireup.c:512
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 3 0x0000000000050c30 uct_iface_invoke_am()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/uct/base/uct_iface.h:523
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 4 0x00000000000554a2 uct_ud_mlx5_iface_poll_rx()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/uct/ib/ud/accel/ud_mlx5.c:415
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 5 0x000000000004d5c2 uct_ud_iface_async_progress()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/uct/ib/ud/base/ud_iface.c:865
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 6 0x000000000003ab2c ucs_async_handler_dispatch()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/ucs/async/async.c:204
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 7 0x000000000003acca ucs_async_dispatch_timerq()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/ucs/async/async.c:260
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 8 0x000000000003d0a8 ucs_async_thread_func()  /hpc/local/benchmarks/hpcx_install_2018-06-02/src/hpcx-gcc-redhat7.4/ucx-master/src/ucs/async/thread.c:123
Sun Jun  3 10:44:38 2018[1,364]<stderr>: 9 0x0000000000007e25 start_thread()  pthread_create.c:0
Sun Jun  3 10:44:38 2018[1,364]<stderr>:10 0x00000000000f834d __clone()  ???:0
Sun Jun  3 10:44:38 2018[1,364]<stderr>:===================-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
---------------------------------------------------------------------------------------------------------------------------------
mpirun noticed that process rank 364 with PID 4124 on node clx-hercules-106 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------+ rc=139
+ exit 139
```

**Comment by amaslenn** [comment]
Couldn't reproduce it in 100 iterations.

**Comment by yosefe** [comment]
according to address, gdb trace is wrong, it's coming from uct_ep_connect_to_ep(), and ep is NULL
```
uct_ep_connect_to_ep():
/hpc/local/benchmarks/hpcx_install_2018-06-03/src/hpcx-gcc-redhat7.4/ucx-master/src/uct/base/uct_iface.c:489
   1fce0:       48 8b 07                mov    (%rdi),%rax  <== here
   1fce3:       48 8b 80 08 01 00 00    mov    0x108(%rax),%rax
   1fcea:       ff e0                   jmpq   *%rax
   1fcec:       0f 1f 40 00             nopl   0x0(%rax)
```

**Comment by yosefe** [comment]
repro after ~300 iters

**Comment by alinask** [comment]
@yosefe closing this one as the fix was merged both to master and to v1.4.
